### PR TITLE
Repair vendor bundles and clean up js library usage

### LINKED ISSUE #1555: Vendor bundles don't export correctly as libraries
This was working but apparently doesn't want to work correctly now. Vendor bundles correctly export to `window`, but are not available via `require()`. This is likely some browserify sloppiness. Vendor libraries should be available on the `global`/`window`, but modular javascript code should expect to use it as a module.

### PR DESCRIPTION
**Pull request by agjohnson**
This repairs vendor bundles, which were not operating as expected as
"standalone" bundles, generated by browserify. The bundles should be imported
globally, but also available as a require() module, but this was not working.
This drops the "standalone" feature for exported modules only, and provides
some more configuration for each entry point.

This also starts to clean up some of the javascript messiness for legacy bundles. This moves the bundles into static collection, symlinks them, and builds standalone bundles which are also based off of bower dependencies. This shouldn't be required, if the standalone bundles worked as expected.

In the future, the doc-embed module could simply be a full module, with dependencies injected as a bundle normally would, though this does nothing to get rid of the legacy bundles in this code, as we'll always have old documentation pointing to the old bundles.

Fixes #1555

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -17,74 +17,94 @@ var gulp = require('gulp'),
...
+    gold: {'js/gold.js': {}},
+    donate: {'js/donate.js': {}}
 };
 
 // Standalone application to create vendor bundles for. These can be imported
 // with require in the browser or with Node during testing.
-var standalone = ['jquery', 'knockout'];
+var standalone = {
+    'jquery': {standalone: 'jquery'},
+    'knockout': {},
```
Comment (ericholscher) [inline gulpfile.js]: Is there a reason this isn't using an explicit standalone like the others?

**Comment by ericholscher** [comment]
:+1: on moving the JS out of the global media path by default. It seems like this cleans up our JS integration a lot, though I don't quite grok everything that is happening.

File: readthedocs/templates/builds/build_detail.html [hunk 2]
```diff
@@ -9,8 +9,9 @@
   <script type="text/javascript" src="{% static 'vendor/knockout.js' %}"></script>
   <script type="text/javascript" src="{% static 'builds/js/detail.js' %}"></script>
   <script type="text/javascript">
+var detail = require('detail');
```
Comment (ericholscher) [inline readthedocs/templates/builds/build_detail.html]: This feels weird. "detail" is quite a generic name.

File: gulpfile.js (hunk 1 above)
Comment (agjohnson) [inline gulpfile.js]: Standalone is a library that functions like the legacy libraries, you can't import via `require()`. Ideally, we'd have one library that could do both, but browserify doesn't create libraries like it suggests.

With a patch to sphinx-ext, these libraries will only ever be used by stale documentation. A patch to sphinx-ext and sphinx-theme will generate libraries that will point to the new libraries that are available by `require()`, or if it makes sense, bundled all together.

I'm playing more with the external javascript pieces today to see what makes the most sense.

Regarding knockoutjs, it is a new dependency and should never be used in the legacy context, always via `require()`.

File: readthedocs/templates/builds/build_detail.html (hunk 2 above)
Comment (agjohnson) [inline readthedocs/templates/builds/build_detail.html]: Perhaps, this could just as easily be `builds/detail`.

Comment (agjohnson) [inline readthedocs/templates/builds/build_detail.html]: Addressed in https://github.com/rtfd/readthedocs.org/commit/26f27ddd77ac745ec060280ae221fdd83bcba8ce

**Pull request merged**